pkg/controllers/dynakube/monitored_entities/reconciler.go

```diff
@@ -0,0 +1,69 @@
+package monitoredentities
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta2/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers"
+)
+
+type ReconcilerBuilder func(
+	dtClient dynatrace.Client,
+	dk *dynakube.DynaKube,
+	kubeSystemUUID string,
+) controllers.Reconciler
+
+func NewReconciler( //nolint
+	dtClient dynatrace.Client,
+	dk *dynakube.DynaKube,
+	kubeSystemUUID string,
+) controllers.Reconciler {
+	return &Reconciler{
+		dk:             dk,
+		dtClient:       dtClient,
+		kubeSystemUUID: kubeSystemUUID,
+	}
+}
+
+type Reconciler struct {
+	dk             *dynakube.DynaKube
+	dtClient       dynatrace.Client
+	kubeSystemUUID string
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context) error {
+	if r.dk.Status.KubernetesClusterMEID == "" {
+		monitoredEntities, err := r.dtClient.GetMonitoredEntitiesForKubeSystemUUID(ctx, r.kubeSystemUUID)
+		if err != nil {
+			log.Info("failed to retrieve MEs")
+
+			return err
+		}
+
+		log.Info("retrieved MEs")
+
+		if len(monitoredEntities) == 0 {
+			log.Info("no MEs found, no monitoredentityID will be set in the dynakube status")
+
+			return nil
+		}
```

Comment:
should we care about the scenario when there is no monitored entity and shouldn't be one? is that a valid scenario ?

because if it is, then we will now "constantly"(ie.: every reconciliation) ping the tenant about it

Reply:
no, you are right, thinking about it this should not be a valid scenario. Will remove this part.

Reply:
if it's not valid, then we should probably error 😅

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/3447/commits/7e1e568e4e9045f69237947d9217d7ec22f8d380